airbyte_cdk/sources/declarative/manifest_declarative_source.py

```diff
@@ -546,6 +546,11 @@ def _dynamic_stream_configs(
                 COMPONENTS_RESOLVER_TYPE_MAPPING[resolver_type],
                 components_resolver_config,
                 config,
+                **(
+                    {"stream_name": dynamic_definition.get("name")}
+                    if resolver_type == "HttpComponentsResolver"
+                    else {}
+                ),
```

Comment:
This part looks a little awkward. We're building a mapping only to immediately de-reference it back into kwargs

It's probably a little more pythonic to just have a normal if/else:
```
if resolver_type == "HttpComponentsResolver":
  components_resolver = self._constructor.create_component(
    model_type=COMPONENTS_RESOLVER_TYPE_MAPPING[resolver_type],
    component_definition=components_resolver_config,
    config=config,
    stream_name=dynamic_definition.get("name"),
  )
else:
  components_resolver = self._constructor.create_component(
    model_type=COMPONENTS_RESOLVER_TYPE_MAPPING[resolver_type],
    component_definition=components_resolver_config,
    config=config,
  )
```